lib/api/api.go

```diff
@@ -363,8 +363,10 @@ func (s *service) Serve(ctx context.Context) error {
 	// Wrap everything in basic auth, if user/password is set.
 	if guiCfg.IsAuthEnabled() {
 		var handlePasswordAuth http.Handler
-		handler, handlePasswordAuth = authAndSessionMiddleware("sessionid-"+s.id.String()[:5], guiCfg, s.cfg.LDAP(), handler, s.evLogger)
+		sessionCookieName := "sessionid-"+s.id.String()[:5]
+		handler, handlePasswordAuth = authAndSessionMiddleware(sessionCookieName, guiCfg, s.cfg.LDAP(), handler, s.evLogger)
 		restMux.Handler(http.MethodPost, "/rest/noauth/auth/password", handlePasswordAuth)
+		restMux.Handler(http.MethodPost, "/rest/noauth/auth/logout", handleLogout(sessionCookieName))
```

Comment:
Why is this method part of `noauth`?

Reply:
Hm, good question.

On the one hand, one probably shouldn't be allowed to terminate other users' sessions without their consent (in general principle, ignoring that Syncthing only supports one user anyway...).

On the other hand, if you don't have a session cookie, then the backend won't actually do anything, as the session to be invalidated is determined by the cookie value. And if you do have a valid session cookie, then by definition you are the owner of that session. So it probably doesn't hurt to have this under `/noauth/`?

But yeah, your question indicates that this violates the principle of least surprise. So I guess I'll move it? :slightly_smiling_face:

Reply:
The least-surprise thing was what triggered my comment in the first place. Maybe there's also value of login/logout being right next to each other. This is one of those things where the longer I think about it the less I have an opinion either way xD

I also had a (anyway very theoretical) attack vector in mind, however writing it out made it disappear: As there's no sleep when one sends an incorrect session id, it's somewhat faster to brute-force it (I know, as I said very theoretical). However if one succeeds, one has the glory of having terminated the session of a user - highly valuable result that is.

Reply:
Alright, I'll add a comment here but leave it as is. Does that sound fair?

Reply:
I've merged this into https://github.com/syncthing/syncthing/pull/8757, let me know there if you'd like any more changes to this.